### ISSUE TITLE
Unexpected number of upserts: 2

### ISSUE THREAD

**Issue by neowu** [bwaldvogel/mongo-java-server#41]
With latest 1.11.0, the bulkWrite with multiple ReplaceOneModel and upsert=true is throwing
"Unexpected number of upserts: xx" exception
(1.9.7 works as expected)

Example Code:
```
List<ReplaceOneModel<T>> models = new ArrayList<>(size);
            for (T entity : entities) {
                Object id = mongo.codecs.id(entity);
                models.add(new ReplaceOneModel<>(Filters.eq("_id", id), entity, new ReplaceOptions().upsert(true)));
            }
            collection().bulkWrite(models, new BulkWriteOptions().ordered(false));
```

and i found mongo-java-server added those lines in 1.11.0
de.bwaldvogel.mongo.backend.AbstractMongoDatabase 
Line 307
```
        if (!upserts.isEmpty()) {
            if (upserts.size() != 1) {
                throw new IllegalStateException("Unexpected number of upserts: " + upserts.size());
            }
            response.put("upserted", upserts);
        }
```
is this expected? by debugging, i think the upserts can be more than one if there are multiple upserts occurred

Thanks

**Comment by bwaldvogel** [comment]
Thanks for reporting this bug. Can you retest on bfc4ae3?

**Comment by neowu** [comment]
Thanks for fixing this so quickly, I tested with latest build, and it fixed my issue.
Please help to schedule release

**Comment by bwaldvogel** [comment]
I’ve just released version `1.11.1` which contains the fix.
